### ISSUE TITLE
add NodeJS API

### ISSUE THREAD

**Issue by ndelangen** [obahareth/are-you-es5#31]
Using this from the command line is great, but we'd like to be able to get the regex from a node process.

Would you be able to tell us how to use this package from a node-process or would you be able to add a feature so we could?

related:
https://github.com/storybookjs/storybook/pull/10725#discussion_r422903309

**Comment by obahareth** [comment]
Hey @ndelangen, does [this comment](https://github.com/obahareth/are-you-es5/issues/25#issuecomment-596087426) from issue #25 help?

**Comment by obahareth** [comment]
@ndelangen I added [this part](https://github.com/obahareth/are-you-es5#usage-as-a-library) to the README, can you let me know if it behaves as you need it to?

I hope I managed to help!

**Comment by ndelangen** [comment]
Thanks, I'll give this a try!

**Comment by tooppaaa** [comment]
Hello @obahareth 
I implemented your solution [here](https://github.com/storybookjs/storybook/pull/10783/commits/8e580d5d796a1724d3523ce44f13415af6a2722b) 

It's working.
In my opinion there are a few steps before being able to use it "in production"

- Usage from `/dist` is not natural. It feels like we're doing something not wanted from the package
- Missing --silent option or similar so we don't output to the client
- `getBabelLoaderIgnoreRegex` is misleading, in my implementation, it returns the string that represents a regexp but I had to remove leading and trailing / to use it as a regexp for my loader.

I can of couse help working on a PR

**Comment by obahareth** [comment]
@tooppaaa You're absolutely right on all your points, I tried to share the quickest working solution I could think of so anyone who needs this at the moment doesn't get blocked. I really would appreciate a PR, because I am swarmed at the moment and I wouldn't be able to implement this soon.

**Comment by tooppaaa** [comment]
How would you like the library to be used ?

My suggestion would be 
```js
import { checkModules, buildIncludeRegexp, buildExcludeRegexp } from "are-you-es5";

const result = checkModules({
    path: "", // Automatically find up package.json from cwd
    checkAllNodeModules: true,
    ignoreBabelAndWebpackPackages: true
});

/**
 * result has props
 * 
 * es6Modules: string[]
 * es5Modules: string[]
 */

 /** Returns the regexp including all es6 modules */
 buildIncludeRegexp(result.es6Modules, {
     webpack: true, // include or not these "fixed" deps by default
     babel: true 
 })

/** Returns the regexp exluding all es6 modules */
buildExcludeRegexp(result.es6Modules)
```

This is a bit different in the current implemention, classes will be hidden, expose only simple functions.

#34 should not be merged as it's depending on these discussions :)

**Comment by obahareth** [comment]
I really like your suggestion but it's a big change, perhaps we could merge #34 as a small improvement to get a better experience for people who need to use this package as a library, and work towards this refactor over the long term?

One question though I'm a little confused as to what the `webpack: true` and `babel: true` mean when you're calling `buildIncludeRegexp`, what's the difference between them and `ignoreBabelAndWebpackPackages: true` when calling `checkModules`? (Although I definitely agree with separating ignore babel/webpack separately)

**Comment by tooppaaa** [comment]
Agree on the options, I get confused I guess.
I can work on this, as it's only a facade of the current implementation, it should be pretty fast. 
If you follow semantic releases, changing the API is a major version, If you merge, I don't think it should be released :)

**Comment by obahareth** [comment]
For #34 I'm not sure if it could be considered a major version as most of the users are using the CLI-version only but you make a good point. It's best to be safe and bump the major version to make sure nobody gets any breaking changes unexpectedly. If you think you could take on the refactor then we can go with that instead of #34.

**Comment by obahareth** [comment]
I will close this since @tooppaaa's work in #34 has refactored the codebase quite a bit and has been released as 2.0.0

Please reopen it if you are still having any issues.
